libs/agno/agno/os/middleware/jwt.py

```diff
@@ -0,0 +1,95 @@
+import jwt
+from typing import Optional, Dict, Any
+from fastapi import HTTPException, Request, Response
+from starlette.middleware.base import BaseHTTPMiddleware
+from agno.utils.log import log_debug
+
+class JWTMiddleware(BaseHTTPMiddleware):
+    """
+    JWT Middleware for extracting user information from JWT tokens.
+    
+    This middleware:
+    1. Extracts JWT token from Authorization header
+    2. Decodes and validates the token
+    3. Extracts user_id and other claims
+    4. Adds them to request state for use in routes
+    """
+    
+    def __init__(
+        self,
+        app,
+        secret_key: str,
+        algorithm: str = "HS256",
+        token_prefix: str = "Bearer",
+        user_id_claim: str = "user_id",
+        auto_error: bool = False,
+    ):
+        super().__init__(app)
+        self.secret_key = secret_key
+        self.algorithm = algorithm
+        self.token_prefix = token_prefix
+        self.user_id_claim = user_id_claim
+        self.auto_error = auto_error
+
+    async def dispatch(self, request: Request, call_next) -> Response:
+        # Extract token from Authorization header
+        authorization: str = request.headers.get("Authorization", "")
+        
+        if not authorization:
+            if self.auto_error:
+                raise HTTPException(status_code=401, detail="Authorization header missing")
+            return await call_next(request)
+
+        try:
+            # Parse token
+            scheme, token = authorization.split(" ", 1)
+            if scheme.lower() != self.token_prefix.lower():
+                if self.auto_error:
+                    raise HTTPException(status_code=401, detail="Invalid authentication scheme")
+                return await call_next(request)
+
+        except ValueError:
+            if self.auto_error:
+                raise HTTPException(status_code=401, detail="Invalid authorization header format")
+            return await call_next(request)
+
+        # Decode JWT token
+        try:
+            payload = jwt.decode(token, self.secret_key, algorithms=[self.algorithm])
+            
+            # Extract user information
+            user_id = payload.get(self.user_id_claim)
+            
+            # Add to request state
+            request.state.jwt_payload = payload
+            request.state.user_id = user_id
```

Comment:
We'll have to do something more clever where we update the actual request arguments, but only for endpoints where it is applicable.
If the endpoint supports `user_id`, we populate it
If the endpoint supports `kwargs`, we send the dependencies there

This would autopopulate requests to /agents/{id}/runs etc